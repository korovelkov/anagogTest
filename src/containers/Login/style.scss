.Login {
  position: relative;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background: white;
  flex-direction: column;
}

.Login__form {
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  margin-top: 30px;
  min-width: 350px;
}

.Login__input-wrapper {
  width: 100%;
  margin-bottom: 20px;
}

.Login__title {
  font-size: 46px;
}

.Login__error {
  color: red;
  padding: 10px;
}
