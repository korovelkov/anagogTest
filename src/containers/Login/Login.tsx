import React, { Component } from 'react';
import { Button, Input } from 'antd';
import { connect } from 'react-redux';

import { login } from '../../redux/user/actions';
import { AppState } from '../../redux';

import './style.scss';

interface IState {
  name: string;
  password: string;
}

interface IProps {
  login: (name: string, password: string) => void;
  name: string;
  error: string;
}

type InputEvent = React.ChangeEvent<HTMLInputElement>;

export class Login extends Component<IProps, IState> {
  state: IState = {
    name: '',
    password: '',
  };

  handleChange = ({ target: { name, value } }: InputEvent): void => {
    this.setState({ [name]: value } as Pick<IState, keyof IState>);
  }

  login = () => {
    const { name, password } = this.state;
    const { login } = this.props;
    login(name, password);
  }

  render() {
    const { error } = this.props;
    const { name, password } = this.state;
    return (
      <div className="Login">
        <h1 className="Login__title">Operational Client</h1>
        <form className="Login__form" noValidate autoComplete="off">
          <div className="Login__input-wrapper">
            <Input
              addonBefore="login"
              id="name"
              name="name"
              className="Login__input"
              value={name}
              onChange={this.handleChange}
              pattern="[:/\\@.;,$#!%^*()_={}[\]~'`+<>|â„–&?]"
            />
          </div>
          <div className="Login__input-wrapper">
            <Input
              addonBefore="password"
              id="password"
              className="Login__input"
              type="password"
              name="password"
              value={password}
              onChange={this.handleChange}

            />
          </div>
          {error && <span className="Login__error">{error}</span>}
          <Button onClick={this.login} type="primary">
            Login
          </Button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = ({ user }: AppState) => ({
  name: user.name,
  error: user.loginError,
});

export default connect(mapStateToProps, { login })(Login);
