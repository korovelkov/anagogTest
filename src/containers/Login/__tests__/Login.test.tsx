import React from 'react';
import { Login } from '../Login';
import renderer from 'react-test-renderer';

it('render Login', () => {
  const tree = renderer.create(<Login login={() => {}} name="name" error={null} />).toJSON();

  expect(tree).toMatchSnapshot();
});
