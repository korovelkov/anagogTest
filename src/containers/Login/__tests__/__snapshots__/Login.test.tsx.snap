// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Login 1`] = `
<div
  className="Login"
>
  <h1
    className="Login__title"
  >
    Operational Client
  </h1>
  <form
    autoComplete="off"
    className="Login__form"
    noValidate={true}
  >
    <div
      className="Login__input-wrapper"
    >
      <span
        className="Login__input ant-input-group-wrapper"
      >
        <span
          className="ant-input-wrapper ant-input-group"
        >
          <span
            className="ant-input-group-addon"
          >
            login
          </span>
          <input
            className="ant-input"
            disabled={false}
            id="name"
            name="name"
            onChange={[Function]}
            onKeyDown={[Function]}
            pattern="[:/\\\\\\\\@.;,$#!%^*()_={}[\\\\]~'\`+<>|â„–&?]"
            style={null}
            type="text"
            value=""
          />
        </span>
      </span>
    </div>
    <div
      className="Login__input-wrapper"
    >
      <span
        className="Login__input ant-input-group-wrapper"
      >
        <span
          className="ant-input-wrapper ant-input-group"
        >
          <span
            className="ant-input-group-addon"
          >
            password
          </span>
          <input
            className="ant-input"
            disabled={false}
            id="password"
            name="password"
            onChange={[Function]}
            onKeyDown={[Function]}
            style={null}
            type="password"
            value=""
          />
        </span>
      </span>
    </div>
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        Login
      </span>
    </button>
  </form>
</div>
`;
